### DISCUSSION TITLE
Introspection

### DISCUSSION THREAD

**Discussion by kevin-j-channon** [carbon-language/carbon-lang#1532] [Language design]
C++ doesn't have introspection and it's not likely to any time soon. Layering this on with something like Carbon could be an incentive to use it, rather than just write the C++ directly.

**Comment by anton-roos** [comment]
Hi, I would like to learn more about introspection. Can you please refer me to a link to read more about this? How is this different from reflection?

**Reply by anton-roos** [reply]
I found this article :) https://en.wikipedia.org/wiki/Type_introspection

**Reply by kevin-j-channon** [reply]
I think "reflection" is the C# term for "introspection" (maybe some other languages too). Personally, I treat them as synonyms, but there may be some subtlety that this misses :)

EDIT: Saw the link to the article. So there you go; not quite synonyms, but one is a subset of the other :)
